### PR TITLE
Draft: make importExportArgs adjust themselves from file path

### PR DESCRIPTION
**Pull request by manavortex**
# make importExportArgs adjust themselves from file path

Not sure if this is a good or a terrible idea

### CODE REVIEW

File: WolvenKit.Common/Model/Arguments/ExportArgs.cs [hunk 1]
```diff
@@ -301,6 +303,17 @@ public class MeshExportArgs : ExportArgs
         /// </summary>
         /// <returns>String</returns>
         public override string ToString() => $"GLTF/GLB | Export Type : {meshExportType} | Lod filter : {LodFilter} | Is Binary : {isGLBinary}";
+
+        public override ImportExportArgs AdjustDefaultFromFilepath(string baseFile)
+        {
+            // We don't want LOD filters to be active by default for weapon meshes 
+            if (baseFile.Contains("weapon", StringComparison.CurrentCultureIgnoreCase))
```
Comment (rfuzzo) [inline WolvenKit.Common/Model/Arguments/ExportArgs.cs:310]: I think this is too dangerous and it's also hidden. 

If we do this it should 
1. be behind a toggle switch in the UI
2. needs some description in the UI
3. needs a warning in the UI that this is active

**Review by rfuzzo — CHANGES_REQUESTED** [review]

File: WolvenKit.Common/Model/Arguments/ExportArgs.cs (hunk 1 above)
Comment (manavortex) [reply WolvenKit.Common/Model/Arguments/ExportArgs.cs:310]: Could absolutely implement it as abstract but I didn't want to change 28 inheriting classes without touching base first :D That would make it not-hidden, at least.

And yeah, I noticed in hindsight that this doesn't only affect the default settings (or does it? This should only become active once, when the object is created…)

**Comment by rfuzzo** [comment]
This can be merged if we show on the UI that the export settings have been set based on some factors.

E.g. lower left-hand corner on the export tool: show an info png ℹ️ with a tooltip "guessing best settings for your file..." if we set the settings based on the filename

**Comment by manavortex** [comment]
If you implement the GUI, I can do the rest.

**Pull request closed without merging**